pkg/registry/generic/etcd/etcd_test.go

```diff
@@ -343,254 +265,106 @@ func TestEtcdUpdate(t *testing.T) {
-			},
-		},
-		E: nil,
-	}
+	testContext := api.WithNamespace(api.NewContext(), "test")
+	server, registry := NewTestGenericEtcdRegistry(t)
+	defer server.Terminate(t)
 
-	nodeWithPodB := tools.EtcdResponseWithError{
-		R: &etcd.Response{
-			Node: &etcd.Node{
-				Value:         runtime.EncodeOrDie(testapi.Default.Codec(), podB),
-				ModifiedIndex: 1,
-				CreatedIndex:  1,
-			},
-		},
-		E: nil,
+	// Test1 try to update a non-existing node
+	_, _, err := registry.Update(testContext, podA)
+	if !errors.IsNotFound(err) {
+		t.Errorf("Unexpected error: %v", err)
 	}
 
-	nodeWithPodAWithResourceVersion := tools.EtcdResponseWithError{
-		R: &etcd.Response{
-			Node: &etcd.Node{
-				Value:         runtime.EncodeOrDie(testapi.Default.Codec(), podAWithResourceVersion),
-				ModifiedIndex: 3,
-				CreatedIndex:  1,
-			},
-		},
-		E: nil,
-	}
-	emptyNode := tools.EtcdResponseWithError{
-		R: &etcd.Response{},
-		E: tools.EtcdErrorNotFound,
+	// Test2 createIfNotFound and verify
+	registry.UpdateStrategy.(*testRESTStrategy).allowCreateOnUpdate = true
+	if !updateAndVerify(t, testContext, registry, podA) {
+		t.Errorf("Unexpected error updating podA")
 	}
+	registry.UpdateStrategy.(*testRESTStrategy).allowCreateOnUpdate = false
 
-	testContext := api.WithNamespace(api.NewContext(), "test")
+	// Test3 outofDate
+	_, _, err = registry.Update(testContext, podAWithResourceVersion)
+	if !errors.IsConflict(err) {
+		t.Errorf("Unexpected error: %v", err)
+	}
 
-	table := map[string]struct {
-		existing                 tools.EtcdResponseWithError
-		expect                   tools.EtcdResponseWithError
-		toUpdate                 runtime.Object
-		allowCreate              bool
-		allowUnconditionalUpdate bool
-		objOK                    func(obj runtime.Object) bool
-		errOK                    func(error) bool
-	}{
-		"normal": {
-			existing: nodeWithPodA,
-			expect:   nodeWithPodB,
-			toUpdate: podCopy(podB),
-			errOK:    func(err error) bool { return err == nil },
-		},
-		"notExisting": {
-			existing: emptyNode,
-			expect:   emptyNode,
-			toUpdate: podCopy(podA),
-			errOK:    func(err error) bool { return errors.IsNotFound(err) },
-		},
-		"createIfNotFound": {
-			existing:    emptyNode,
-			toUpdate:    podCopy(podA),
-			allowCreate: true,
-			objOK:       hasCreated(t, podA),
-			errOK:       func(err error) bool { return err == nil },
-		},
-		"outOfDate": {
-			existing: newerNodeWithPodA,
-			expect:   newerNodeWithPodA,
-			toUpdate: podCopy(podB),
-			errOK:    func(err error) bool { return errors.IsConflict(err) },
-		},
-		"unconditionalUpdate": {
-			existing:                 nodeWithPodAWithResourceVersion,
-			allowUnconditionalUpdate: true,
-			toUpdate:                 podCopy(podA),
-			objOK:                    func(obj runtime.Object) bool { return true },
-			errOK:                    func(err error) bool { return err == nil },
-		},
+	// Test4 normal update and verify
+	if !updateAndVerify(t, testContext, registry, podB) {
```

Comment:
Is it necessary for podB to have RV=3? Can't we remove it? If not can you please add a comment?